src/ViewModels/ShellViewModel.cs

```diff
@@ -54,7 +55,13 @@ public INavigationViewService NavigationViewService
 
     public async Task OnLoaded()
     {
-        if (await _localSettingsService.ReadSettingAsync<bool>(WellKnownSettingsKeys.IsNotFirstRun))
+        // Load the Machine Configuration Page if the app is launched via file activation
+        var activatedEventArgs = AppInstance.GetCurrent().GetActivatedEventArgs();
+        if (activatedEventArgs.Kind == ExtendedActivationKind.File)

```

Comment:
This is fine for now, but if we ever handle other file types as well this will need to be updated.